### PR TITLE
feat(schema): GTM server side data transform for schema event_v2

### PR DESCRIPTION
**Pull request by amliuyong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Review by zxkane — COMMENTED** [review]
Is file `src/data-pipeline/spark-etl/src/test/resources/gtm-server/etl_gtm_user_v2_props_incremental_v1/update_date=20240312/app_id=GMTServerLogin2/part-00000-8f5a4cb2-f6aa-4b4f-8c6a-08145a4e5b15.c000.snappy.parquet` used in the test cases?

File: src/data-pipeline/spark-etl/lombok.config [hunk 1]
```diff
@@ -1,2 +1,2 @@
-lombok.setter.flagUsage=error
+#lombok.setter.flagUsage=error
```
[the remark this answers is not in the document]
Comment (amliuyong) [reply src/data-pipeline/spark-etl/lombok.config:1]: removed

**Comment by amliuyong** [comment]
> Is file `src/data-pipeline/spark-etl/src/test/resources/gtm-server/etl_gtm_user_v2_props_incremental_v1/update_date=20240312/app_id=GMTServerLogin2/part-00000-8f5a4cb2-f6aa-4b4f-8c6a-08145a4e5b15.c000.snappy.parquet` used in the test cases?

yes.

**Comment by amliuyong** [comment]
I push new commits, but this PR is not updated. will closed this PR and open a new one.

**Pull request closed without merging**